### PR TITLE
Include on-premise documentation as part of general Sigrid documentation

### PR DESCRIPTION
**Pull request by dennis-sig**

### CODE REVIEW

File: docs/organization-integration/onpremise-integration.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+- Your infrastructure needs to support running applications on Kubernetes.
+- You need to pull Sigrid container images from DockerHub.
+  - If you do not allow outbound internet access, a custom air-gapped installation and update procedure needs to be designed, documented and followed.
+- You need to support and manage a Postgres database service.
+- You need to have a development platform with a continuous integration pipeline, since Sigrid will integrate with this pipeline. Supported development platforms are listed in the "Sigrid CI pipeline integration" in this documentation.
+- You need to have an identity provider supporting OpenID Connect, since Sigrid will integrate with this for authentication.
+- You need to allow outbound internet traffic in order to support Sigrid’s Open Source Health feature. Sigrid needs to connect to external sources to check for the latest vulnerability data.
+  - If you do not allow outbound internet access, the Open Source Health feature is not available. The rest of Sigrid is unaffected.
+- For viewing source code within Sigrid, you need to provide a development platform that is integrated with th esame identity provider as Sigrid itself.
```
Comment (fabianl-sig) [inline docs/organization-integration/onpremise-integration.md:34]:
```suggestion
- For viewing source code within Sigrid, you need to provide a development platform that is integrated with the same identity provider as Sigrid itself.
```

File: docs/organization-integration/onpremise-integration.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+- Authentication is based on your identity provider, using [OpenID Connect](https://openid.net/developers/how-connect-works/). 
+- Analyses are triggered from a build pipeline. The analysis results are then imported into a Postgres database, so they can be viewed in Sigrid.
+- Large files are stored in an [S3-compatible object store](https://aws.amazon.com/s3/).
+
+Some Sigrid on-premise features are *optional* based on your environment:
+
+- The Open Source Health feature requires outbound internet access, to retrieve the latest version of all known vulnerabilities in open source libraries. 
+- When viewing detailed analysis results, Sigrid displays relevant source code files within Sigrid. For this to work, a web-accessible code storage needs to be available. This integrates with Sigrid via [OAuth](https://oauth.net/2/). For this to work, the identity provider used for Sigrid authentication and for the code storage needs to be the same.
+
+### Environment requirements
```
Comment (fabianl-sig) [inline docs/organization-integration/onpremise-integration.md:24]: Do we want to specify versions? For example we have a minimum version requirement of kubernetes version 1.21  and Postgres version 13 but only aim to support non EOL versions of kubernetes (the last three releases)?

File: docs/organization-integration/onpremise-integration.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+## Obtaining Sigrid on-premise
+
+The Docker containers that form Sigrid on-premise are distributed via [DockerHub](https://hub.docker.com). You will receive an account that allows you to access the container registry. 
+
+<img src="../images/onpremise-dockerhub.png" width="500" /> 
+
+As explained above, Sigrid consists of several Docker containers. The container `sigrid-multi-analysis-import` will run directly from your development platform, all other containers are deployed to your Kubernetes cluster. These steps are explained in more detail in the following sections.
+
+### Updating Sigrid on-premise to a new version
```
Comment (fabianl-sig) [inline docs/organization-integration/onpremise-integration.md:45]: It is important that all components (APIs and analysis container) are update to the same version.

**Review by fabianl-sig — CHANGES_REQUESTED** [review]

File: docs/organization-integration/onpremise-integration.md (hunk 3 above)
Comment (dennis-sig) [reply docs/organization-integration/onpremise-integration.md:45]: Good one, will add.

File: docs/organization-integration/onpremise-integration.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  stage: test
+  image: softwareimprovementgroup/sigrid-multi-analysis-import
+  entrypoint: [""]
+  variables:
+    CUSTOMER: YOUR_SIGRID_CUSTOMER
+    SYSTEM: YOUR_SIGRID_SYSTEM
+    POSTGRES_HOST_AND_PORT: YOUR_DATABASE_HOST:5432
+    POSTGRES_PASS: $SIGRID_REVIEW_ONPREM_POSTGRES_IMPORTER_PASS
+  script:
+    - ln -s $PWD /tmp/sources
```
Comment (tukraus) [inline docs/organization-integration/onpremise-integration.md:122]: Same here, maybe we can insert this in the all-the-things.sh script (without breaking if the symlink is already existing)

File: docs/organization-integration/onpremise-integration.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+      - sigridci/index.html
+      - sigridci/feedback.md
+    expire_in: 1 week
+    when: always
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+```
+
+- This example assumes your pipeline includes a stage called `test`. If your pipeline is structured differently you will need to replace this with the name of the appropriate stage where you want Sigrid CI to run. 
+- Replace `YOUR_SIGRID_CUSTOMER` and `YOUR_SIGRID_SYSTEM` with your Sigrid customer account name and system name respectively. If you're not sure about your Sigrid customer account name, ask your Sigrid administrator. You can choose your own system name, though system names can only contain letters, numbers, and hyphens.
```
Comment (tukraus) [inline docs/organization-integration/onpremise-integration.md:138]:
```suggestion
- Replace `YOUR_SIGRID_CUSTOMER` and `YOUR_SIGRID_SYSTEM` with your Sigrid customer account name and system name, respectively. If you're not sure about your Sigrid customer account name, ask your Sigrid administrator. You can choose your own system name, though system names can only contain letters, numbers, and hyphens.
```

File: docs/organization-integration/onpremise-integration.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+Next, you will need to schedule *when* you're going to publish your system to Sigrid. Azure DevOps allows you to [schedule pipelines](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/scheduled-triggers?view=azure-devops&tabs=yaml). You can use this to periodically publish your system to Sigrid, for example as a nightly build. Alternatively, you can use Sigrid CI on a more continuous basis by running the pipeline step after every commit.
+
+## Functional differences in Sigrid on-premise
+
+- The on-premise Sigrid distribution is single tenant. You cannot create your own "tenants", all systems and analyses will end up in your portfolio. That said, you can still use Sigrid’s user management to define which people should have access to which systems.
+- [Mendix Quallty and Security Management](https://www.softwareimprovementgroup.com/mendix-quality-and-security-management-powered-by-sigrid/) is only available if you are using Mendix in combination with GitHub or GitLab.
```
Comment (tukraus) [inline docs/organization-integration/onpremise-integration.md:175]:
```suggestion
- [Mendix Quality and Security Management](https://www.softwareimprovementgroup.com/mendix-quality-and-security-management-powered-by-sigrid/) is only available if you are using Mendix in combination with GitHub or GitLab.
```

**Review by tukraus — CHANGES_REQUESTED** [review]
Mostly minor suggestions

File: docs/organization-integration/onpremise-integration.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+### Environment requirements
+
+- Your infrastructure needs to support running applications on Kubernetes.
+- You need to pull Sigrid container images from DockerHub.
+  - If you do not allow outbound internet access, a custom air-gapped installation and update procedure needs to be designed, documented and followed.
+- You need to support and manage a Postgres database service.
+- You need to have a development platform with a continuous integration pipeline, since Sigrid will integrate with this pipeline. Supported development platforms are listed in the "Sigrid CI pipeline integration" in this documentation.
+- You need to have an identity provider supporting OpenID Connect, since Sigrid will integrate with this for authentication.
+- You need to allow outbound internet traffic in order to support Sigrid’s Open Source Health feature. Sigrid needs to connect to external sources to check for the latest vulnerability data.
```
[the remark this answers is not in the document]
Comment (dennis-sig) [reply docs/organization-integration/onpremise-integration.md:32]: Interestingly, both originate from the original Word document that we used before moving this into the Sigrid documentation :) But you're right, let's consolidate it.

File: docs/organization-integration/onpremise-integration.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  sigridci:
+    container:
+      image: softwareimprovementgroup/sigrid-multi-analysis-import
+      env:
+        CUSTOMER: YOUR_SIGRID_CUSTOMER
+        SYSTEM: YOUR_SIGRID_SYSTEM
+        POSTGRES_HOST_AND_PORT: YOUR_DATABASE_HOST:5432
+        POSTGRES_PASS: $SIGRID_REVIEW_ONPREM_POSTGRES_IMPORTER_PASS
+    steps:
+      - ln -s $PWD /tmp/sources
```
[the remark this answers is not in the document]
Comment (tukraus) [reply docs/organization-integration/onpremise-integration.md:85]: So for github/gitlab/azure devops the symlink creation isn't necessary at all anymore, but we can also take this out later.

File: docs/organization-integration/onpremise-integration.md [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+- The Open Source Health feature requires outbound internet access, to retrieve the latest version of all known vulnerabilities in open source libraries. 
+- When viewing detailed analysis results, Sigrid displays relevant source code files within Sigrid. For this to work, a web-accessible code storage needs to be available. This integrates with Sigrid via [OAuth](https://oauth.net/2/). For this to work, the identity provider used for Sigrid authentication and for the code storage needs to be the same.
+
+### Environment requirements
+
+- Your infrastructure needs to support running applications on Kubernetes.
+- You need to pull Sigrid container images from DockerHub.
+  - If you do not allow outbound internet access, a custom air-gapped installation and update procedure needs to be designed, documented and followed.
+- You need to support and manage a Postgres database service.
+- You need to have a development platform with a continuous integration pipeline, since Sigrid will integrate with this pipeline. Supported development platforms are listed in the "Sigrid CI pipeline integration" in this documentation.
```
[the remark this answers is not in the document]
Comment (tukraus) [reply docs/organization-integration/onpremise-integration.md]: no longer relevant

**Pull request merged**